Automattic/sensei
# Add REST API for Onboarding user data

Fixes #3048

### Changes proposed in this Pull Request

* Add methods to store and retrieve user data input during onboarding as an option `sensei_setup_wizard_data`, with fields for `purpose`, `features`.
* Add a single endpoint for getting data for the Setup Wizard frontend. 
* Add REST API endpoints for Purpose and Features steps that save their data.
* Add schema validation to all Setup Wizard REST API endpoints.
* Hook up Purpose API to frontend. 
* Add tracking of completed steps.  
* The submit features endpoint has a plugin whitelist, to be loaded from the same source as the list of displayed plugins.
 
### Testing instructions

* User input on the Purpose step should be persisted now.
* Attempting to call the `purpose` API endpoint with purpose slugs other than the whitelist should result in error. 
* Attempting to call any of the REST API endpoints without the required parameters should result in error.
